Content.Shared/_RMC14/Medical/MedivacStretcher/SharedMedivacStretcherSystem.cs

```diff
@@ -0,0 +1,280 @@
+        var slot = _container.EnsureContainer<ContainerSlot>(ent.Owner, MedivacStretcherComponent.BuckledSlotId);
+        if (slot.Count > 0)
+        {
+            args.Verbs.Add(new InteractionVerb()
+            {
+                Text = Loc.GetString("verb-categories-unbuckle"),
+                Icon = new SpriteSpecifier.Texture(new("/Textures/Interface/VerbIcons/unbuckle.svg.192dpi.png")),
+                Act = () =>
+                {
+                    Unstrap(ent);
+                },
+                Priority = 1
+            });
+        }
+    }
+
+    private void OnFold(Entity<MedivacStretcherComponent> ent, ref FoldedEvent args)
+    {
+        if (args.IsFolded)
+        {
+            DeactivateBeacon(ent.Owner);
+        }
+    }
+
+    private void PrepareMedivac(Entity<MedivacStretcherComponent> ent, ref PrepareMedivacEvent args)
+    {
+        if (!TryComp(ent.Owner, out StrapComponent? strapComp))
+        {
+            return;
+        }
+
+        var slot = _container.EnsureContainer<ContainerSlot>(ent.Owner, MedivacStretcherComponent.BuckledSlotId);
+
+        if (slot.ContainedEntity is not { } buckled)
+        {
+            return;
+        }
+        _appearance.SetData(ent.Owner, MedivacStretcherVisuals.MedivacingState, true);
+        args.ReadyForMedivac = true;
+    }
+
+    private void OnInteract(Entity<MedivacStretcherComponent> ent, ref InteractHandEvent args)
+    {
+        if (args.Handled)
+        {
+            return;
+        }
+
+        if (TryComp(ent.Owner, out FoldableComponent? foldComp) &&
+            foldComp.IsFolded)
+        {
+            return;
+        }
+
+        ToggleBeacon(args.Target, args.User);
+        args.Handled = true;
+    }
+
+    private void OnTryStrap(Entity<MedivacStretcherComponent> ent, ref StrapAttemptEvent args)
+    {
+        var slot = _container.EnsureContainer<ContainerSlot>(ent.Owner, MedivacStretcherComponent.BuckledSlotId);
+        _container.Insert(args.Buckle.Owner, slot);
+        _appearance.SetData(ent.Owner, StrapVisuals.State, true);
+
+        if (!TryComp(ent.Owner, out StrapComponent? strapComp))
+        {
+            return;
+        }
+        _audio.PlayPredicted(strapComp.BuckleSound, args.Strap.Owner, args.User);
+        args.Cancelled = true;
+    }
+
+    private void ToggleBeacon(EntityUid stretcher, EntityUid user)
+    {
+        if (HasComp<DropshipTargetComponent>(stretcher))
+        {
+            DeactivateBeacon(stretcher);
+        }
+        else
+        {
+            ActivateBeacon(stretcher, user);
+        }
+    }
+
+    private bool ActivateBeacon(EntityUid stretcher, EntityUid user)
+    {
+        if (HasComp<DropshipTargetComponent>(stretcher))
+        {
+            return true;
+        }
+        EntityCoordinates stretcherCoords = stretcher.ToCoordinates();
+        if (!_dropshipWeaponSystem.CasDebug &&
+            !_areas.CanCAS(stretcher.ToCoordinates().SnapToGrid(_entites, _mapManager)))
```

Comment:
This'll probably be caught in review but you shouldn't use CanCAS here. Like the mortar system you should get the area and check the Medevac field.